bokeh/server/views/ws.py

```diff
@@ -158,6 +161,14 @@ def select_subprotocol(self, subprotocols):
         self._token = subprotocols[1]
         return subprotocols[0]
 
+    def get_compression_options(self):
+        if self._compression_level is None:
+            return None
+        options = {'compression_level': self._compression_level}
+        if self._mem_level is not None:
+            options['mem_level'] = self._mem_level
+        return options
```

Comment:
@philippjfr  is this function necessary? It does not appear to be used anywhere

Reply:
This is the API Tornado defines to enable/disable websocket compression, see [the Tornado docs](https://www.tornadoweb.org/en/stable/websocket.html#tornado.websocket.WebSocketHandler.get_compression_options).